src/core/stream/period/get_adaptation_switch_strategy.ts

```diff
@@ -153,8 +184,11 @@ export default function getAdaptationSwitchStrategy(
   }
 
   const toRemove = excludeFromRanges(unwantedRange, rangesToExclude);
+  const strategyName = adaptation.type === "audio" && audioTrackSwitchingMode === "direct"
+                   ? "needs-buffer-flush"
```

Comment:
A flush should only happen if we're currently playing the Period, no?
There should even be no need if we're playing far in totally another Period

Reply:
Here you check that there are segments already buffered in the current Period which is not the same thing as knowing whether we are playing the current Period.

I propose that we do the flush only if we are in or close to the wanted period. Something like:
```js
      // Check that we're currently playing in or very close to the current Period 
      isTimeInRange({ start: start - 1, end: end + 1 }, currentTime)
```
Here `start` is `period.start` (already deconstructed) and period is `period.end ?? Infinity` (also already defined).
Removing and adding 1 second respectively to the start and the end would allow doing the flush in the case segments overlap a little.